src/modules/global/components/RouteRootGuard.tsx

```diff
@@ -0,0 +1,54 @@
+import { PropsWithChildren, useEffect, useState } from "react";
+import { onAuthStateChanged } from "firebase/auth";
+import { auth } from "../../firebase";
+import { loggedState } from "../state";
+import { useSetRecoilState, useRecoilState } from "recoil";
+import { userState } from "../../authentification/state";
+import { configuratorRoutes } from "../../configurator/const";
+import { Navigate } from "react-router-dom";
+import { PageLoading } from "./PageLoading";
+import { NotFoundPage } from "../../../shared/NotFoundPage";
+
+interface Props extends PropsWithChildren {
+  isErrorPage?: boolean;
+}
+
+export const RouteRootGuard = ({ children, isErrorPage }: Props) => {
+  const [isLoggedIn, setIsLoggedIn] = useRecoilState(loggedState);
+  const setUserData = useSetRecoilState(userState);
+  const [isPageLoading, setIsPageLoading] = useState(true);
+
+  useEffect(() => {
```

Comment:
Remove this because you have the provider now.

Reply:
Done